### PR TITLE
Add option to request a stable amount instead of the request token.

### PR DESCRIPTION
**Pull request by willjgriff**
Includes tests.

Returned amount will always be in the main request token.

### CODE REVIEW

File: test/conviction-voting.js [hunk 1]
```diff
@@ -951,14 +986,52 @@ contract('ConvictionVoting', ([appManager, user, beneficiary]) => {
...
+            const beneficiaryBalanceAfter = await requestToken.balanceOf(beneficiary)
+            assert.equal(beneficiaryBalanceAfter.toNumber(), beneficiaryBalanceBefore.toNumber() + requestTokenReturnAmount, 'Incorrect beneficiary balance')
+          })
+        })
+      })
+    })
   })
 
-  context('Pure functions', () => {
+  context.only('Pure functions', () => {
```
Comment (fabriziovigevani) [inline test/conviction-voting.js:1029]: Might forgot to remove 😄 
```suggestion
  context('Pure functions', () => {
```

File: test/conviction-voting.js [hunk 2]
```diff
@@ -951,14 +986,52 @@ contract('ConvictionVoting', ([appManager, user, beneficiary]) => {
...
+          await convictionVoting.stakeToProposal(proposalId, stakeAmount)
+        })
+
+        context('executeProposal(proposalId)', () => {
+          it('converts stable amount to request token and transfers to beneficiary', async () => {
+            const vaultBalanceBefore = await requestToken.balanceOf(vault.address)
+            const beneficiaryBalanceBefore = await requestToken.balanceOf(beneficiary)
+            const requestTokenReturnAmount = stableRequestedAmount / FEE_TOKEN_PRICE_IN_STABLE_TOKEN
+            const stakeAmount = 10000
+            await convictionVoting.stakeToProposal(proposalId, stakeAmount)
```
Comment (fabriziovigevani) [inline test/conviction-voting.js:1014]: I think this is not needed as we already staked on the `beforeEach`

**Review by fabriziovigevani — COMMENTED** [review]
Nice man!

Some comments and suggestions.

I like the stable request amount  being optional.

File: test/conviction-voting.js (hunk 1 above)
Comment (fabriziovigevani) [reply test/conviction-voting.js:1029]: Never mind, just saw it updated on next PR

File: contracts/ConvictionVoting.sol [hunk 3]
```diff
@@ -227,11 +249,16 @@ contract ConvictionVoting is DisputableAragonApp, TokenManagerHook {
...
+        require(proposal.convictionLast > calculateThreshold(proposal.requestedAmount, proposal.stableRequestAmount), ERROR_INSUFFICIENT_CONVICION);
 
         proposal.proposalStatus = ProposalStatus.Executed;
         _closeDisputableAction(proposal.agreementActionId);
-        vault.transfer(requestToken, proposal.beneficiary, proposal.requestedAmount);
+
+        uint256 requestedAmount = proposal.requestedAmount;
+        if (proposal.stableRequestAmount) {
+            requestedAmount = stableTokenOracle.consult(stableToken, requestedAmount, requestToken);
+        }
```
[the remark this answers is not in the document]
Comment (willjgriff) [reply contracts/ConvictionVoting.sol]: I have done the above. However, it means the front-end will have to convert the stable amount to the request amount using the oracle when calculating the threshold displayed to users creating new proposals.

[the remark this answers is not in the document]
Comment (fabriziovigevani) [reply contracts/ConvictionVoting.sol]: Yeah i think that's fine. We are currently calculating the threshold directly from the frontend using the same math the contract does.

**Review by fabriziovigevani — APPROVED** [review]
LGTM!

**Pull request merged**